lib/perl/Genome/Site/TGI/Extension/UnarchiveInstrumentData.pm

```diff
@@ -0,0 +1,196 @@
+package Genome::Config::AnalysisProject::Command::UnarchiveInstrumentData;
+
+use strict;
+use warnings;
+
+use Genome;
+use Genome::Config::AnalysisProject::Command::UnarchiveInstrumentData; #load real module first
+
+use File::Basename qw();
+use File::Spec qw();
+
+use Sub::Install;
+
+Sub::Install::reinstall_sub ({
+    as => 'unarchive_additional_data',
+    code => \&unarchive_lims_data,
+});
+
+__PACKAGE__->__meta__->property('mount_path')->is_optional(0);
+
+sub unarchive_lims_data {
+    my $self = shift;
+    my $instrument_data = shift;
+
+    my $lims_unarchive_script_path = $self->resolve_lims_unarchive_script_path();
+
+    my $dv = $self->mount_path;
+    if ($instrument_data->class ne 'Genome::InstrumentData::Solexa') { return 1; }
+
+    my $bam_path = $instrument_data->bam_path;
+    if (-e $bam_path) { return 1; }
+
+    $self->status_message('Working on missing BAM %s for instrument data %s', $bam_path, $instrument_data->id);
+
+    my ($bam_filename, $bam_dirname) = File::Basename::fileparse($bam_path);
+    $bam_dirname =~ s/\/$//;
+    my @bam_dirnames = File::Spec->splitdir($bam_dirname);
+
+    my ($unarchived_bam_path) = glob(File::Spec->join($dv->mount_path,'*','csf_*',$bam_filename));
```

Comment:
See above comment on line36.  Perhaps the `csf_*` part of the glob is not necessary?